quartz_solar_forecast/forecast.py

```diff
@@ -30,6 +22,69 @@ def run_forecast(site: PVSite, ts: datetime | str = None, nwp_source: str = "ico
     pv_xr = make_pv_data(site=site, ts=ts)
 
     # load and run models
-    pred_df = forecast_v1(nwp_source, nwp_xr, pv_xr, ts)
+    pred_df = forecast_v1(nwp_source, nwp_xr, pv_xr, ts, model=model)
 
     return pred_df
+
+
+def predict_tryolabs(
+    site: PVSite, ts: datetime | str = None):
+    """Run the forecast with the tryolabs model"""
+
+    solar_power_predictor = TryolabsSolarPowerPredictor()
+
+    solar_power_predictor.load_model()
+    
+    if ts is None:
+        start_date = pd.Timestamp.now().strftime("%Y-%m-%d")
+        start_time = pd.Timestamp.now().floor("15min")
+    else:
+        start_date = pd.Timestamp(ts).strftime("%Y-%m-%d")
+        start_time = pd.Timestamp(ts).floor("15min")
+
+    end_time = start_time + pd.Timedelta(hours=48)
+
+    predictions = solar_power_predictor.predict_power_output(
+        latitude=site.latitude,
+        longitude=site.longitude,
+        start_date=start_date,
+        kwp=site.capacity_kwp,
+        orientation=site.orientation,
+        tilt=site.tilt,
+    )
+
+    if predictions is not None:
+        predictions = predictions[
+            (predictions["date"] >= start_time) & (predictions["date"] < end_time)
+        ]
+        predictions = predictions.reset_index(drop=True)
+        predictions.set_index("date", inplace=True)
+        print("Predictions finished.")
+        return predictions
+
+
+def run_forecast(
+    site: PVSite,
+    model: str = "ocf",
+    ts: datetime | str = None,
+    nwp_source: str = "icon",
+) -> pd.DataFrame:
+    """
+    Predict solar power output for a given site using a specified model.
+
+    :param site: the PV site
+    :param model: the model to use for prediction, choose between "ocf" and "tryolabs",
+                    by default "tryolabs" is used
+    :param ts: the timestamp of the site. If None, defaults to the current timestamp rounded down to 15 minutes.
+    :param nwp_source: the nwp data source. Either "gfs" or "icon". Defaults to "icon" 
+                       (only relevant if model=="ocf")
+    :return: The PV forecast of the site for time (ts) for 48 hours
+    """
+
+    if model == "ocf":
+        return predict_ocf(site, None, ts, nwp_source)
+              
+    if model == "tryolabs":
```

Comment:
could you do `if` and `elif` and add a warning if model no in "ocf" or "tryolabs"